src/common/utils/account.ts

```diff
@@ -0,0 +1,45 @@
+import { flags } from '@oclif/command';
+import { AxiosInstance } from 'axios';
+import inquirer from 'inquirer';
+
+export interface Account {
+  id: string;
+  name: string;
+}
+
+export class AccountUtils {
+  static flags = {
+    account: flags.string({
+      description: 'Architect Account',
+      env: 'ARCHITECT_ACCOUNT',
+      char: 'a',
+    }),
+  };
+
+  static async getAccount(api: AxiosInstance, account_name?: string): Promise<Account> {
+    let account: Account;
+    if (account_name) {
+      account = (await api.get(`/accounts/${account_name}`)).data;
+    } else {
+      inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+      let accounts: Account[] = [];
+      const answers: any = await inquirer.prompt([
+        {
+          type: 'autocomplete',
+          name: 'account',
+          message: 'Select an account',
+          filter: (x) => x, // api filters
+          source: async (answers_so_far: any, input: string) => {
```

Comment:
Is this supposed to be paginated automatically? This currently just cycles through the same ten choices. I assume it's the same for the other utils

Reply:
It doesn't support pagination unfortunately - you can filter down by typing.

At some point we can try to add pagination, but it hopefully won't be a big deal early on.
I don't expect people to have a lot of accounts/platforms - maybe envs
https://github.com/mokkabonna/inquirer-autocomplete-prompt

Reply:
Can we include in the message for each one of these classes something like "Select an account, type to search" or something then? When someone hits the up or down arrow, the prompt "(Use arrow keys or type to search)" disappears and I can see this being confusing for some users. The only instructions left are to use the arrow keys to move up or down

![Screenshot from 2020-08-05 12-58-56](https://user-images.githubusercontent.com/8558567/89442487-ee94b280-d71c-11ea-89ad-88d699473df6.png)
![Screenshot from 2020-08-05 12-59-09](https://user-images.githubusercontent.com/8558567/89442498-f48a9380-d71c-11ea-84db-6b452c150c3c.png)

Reply:
Yeah its not ideal. I was trying to avoid forking/writing our own, but feel free to story/prioritize improvements

Reply:
Looked into it and in order to make a change to that actual line we'd have to fork the autocomplete module to update the Paginator used [here](https://github.com/mokkabonna/inquirer-autocomplete-prompt/blob/master/index.js#L14), then probably copy [this file](https://github.com/SBoudrias/Inquirer.js/blob/132682ecfb3972b0efc9b11a370e649a4a8b49c6/packages/inquirer/lib/utils/paginator.js#L42) into the forked module to update the text. Seems excessive for a text change but also pretty straightforward. On the other hand I didn't think including a little instruction in the prompt message was that bad. I just want to avoid people getting frustrated